entrypoint.rb

```diff
@@ -1,31 +1,41 @@
-require 'openai'
-require 'dotenv/load'
-require 'json'
-require 'pry'
-require_relative 'app/open_ai_client'
-require_relative 'app/submission'
-require_relative 'app/pupilfirst_api'
+require "openai"
+require "dotenv/load"
+require "pry"
+require_relative "app/open_ai_client"
+require_relative "app/submission"
+require_relative "app/pupilfirst_api"
+require_relative "app/reviewer"
 
 OpenAI.configure do |config|
-    config.access_token = ENV.fetch("OPENAI_ACCESS_TOKEN")
-    config.organization_id = ENV.fetch("OPENAI_ORGANIZATION_ID", "") # Optional.
+  config.access_token = ENV.fetch("OPENAI_ACCESS_TOKEN")
+  config.organization_id = ENV.fetch("OPENAI_ORGANIZATION_ID", "") # Optional.
 end
 
 def generate_response
-    @generate_response ||=
-    begin
-        JSON.parse(OpenAIClient.new.ask)
-    rescue => exception
-        {
-            status: "skip",
-            feedback: "Failure message"
-        }
-    end
+  retries = 0
+  generate_response = OpenAIClient.new.ask
```

Comment:
Not a good idea to have a variable name with the same name as the method its defined in.